### DISCUSSION TITLE
Solved: There is no user connected to Revit. and not found method from different class

### DISCUSSION THREAD

**Discussion by mahmoudshaddad** [ricaun-io/RevitTest#12] [Q&A] [answered]
i wrote a simple code to test it open it Revit it show the tool but in VS it show me faild with  There is no user connected to Revit.
![image](https://github.com/user-attachments/assets/67babfd6-f358-4112-a686-8b5a2ef5f590)

**Comment by ricaun** [comment, marked as answer]
You need to have an Autodesk user connected, without an account connect the RevitTest ignore all tests.

**Reply by leomalyshev** [reply]
@ricaun please tell me, is there a way to remove the connection to the account so that the tests work offline?

**Reply by ricaun** [reply]
You need to have an Autodesk user connected inside Revit, is some of the requirements to run the RevitTest plugin.

**Comment by mahmoudshaddad** [comment]
i fixe this but face a different problem bz calling my function 
it shows me the following 
   System.MethodAccessException: Attempt by method 'My Unittest class ' to access method 'my traget method that i want to test' **failed**.
   while there is no error from VS 
  I get the error only while testing the i cannot call my function

**Reply by ricaun** [reply]
Could you give an example of a test?
Maybe you already have old version already loaded in Revit and that version is used instead of the new one with the method available.

**Reply by mahmoudshaddad** [reply]
![Screenshot 2024-08-30 150401](https://github.com/user-attachments/assets/15cf3642-0aa7-4f08-bfeb-5b1611822d3b) 
while Getwallthickness is simple function 
  public int GetWallThickness(Wall wall)
  {
      return (int)Math.Round(UnitUtils.ConvertFromInternalUnits(wall.Width, MillimetersUnitType));
  }
 no error showing in VS

**Reply by mahmoudshaddad** [reply]
As you can see no issues found 
+ i added the file as a  Project not as a dll file

**Reply by ricaun** [reply]
Where this `getWallData` comes from? Is your main addin?
I suspect that your Revit have your addin already loaded, the Test gonna use that assembly already loaded to run the test. And that version probably does not have the method `GetWallThickness`.

**Reply by mahmoudshaddad** [reply]
I had deleted my DLL from the Addin folder, I also removed the Revit API and Revit API UI DLLs. I then cleaned my solution and attempted to rebuild it, but this _did not resolve the issue._

However, after further investigation, I was able to fix the problem by:

Aligning the` processor architecture to x64, `to the same process 
Updating both` System.Runtime.CompilerServices.Unsafe` and `Microsoft.Bcl.AsyncInterfaces.`

 Thank you Ricaun for your replies and your tools
